File: 2696. Minimum String Length After Removing Substrings/solution.py
class Solution(object):
  def minLength(self, s):
    """
    :type s: str
    :rtype: int
    """
    stack = []
    for letter in s:
      if letter == 'B' and len(stack) > 0 and stack[-1] == 'A':
        stack.pop()
      elif letter == 'D' and len(stack) > 0 and stack[-1] == 'C':
        stack.pop()
      else:
        stack.append(letter)
    return len(stack)

File: 2696. Minimum String Length After Removing Substrings/solution.java
class Solution {
  public int minLength(String s) {
    Stack<Character> stack = new Stack<Character>();
    for (int index = 0; index < s.length(); index++){
      Character letter = s.charAt(index);
      if (letter == 'B' && stack.size() > 0 && stack.peek() == 'A'){
        stack.pop();
      }
      else if (letter == 'D' && stack.size() > 0 && stack.peek() == 'C'){
        stack.pop();
      }
      else{
        stack.push(letter);
      }
    }
    return stack.size();
  }
}